sql/parser/stmt.go

```diff
@@ -34,42 +34,125 @@ import "fmt"
-func (*BeginTransaction) statement()    {}
-func (*CommitTransaction) statement()   {}
-func (*CreateDatabase) statement()      {}
-func (*CreateIndex) statement()         {}
-func (*CreateTable) statement()         {}
-func (*Delete) statement()              {}
-func (*DropDatabase) statement()        {}
-func (*DropIndex) statement()           {}
-func (*DropTable) statement()           {}
-func (*Explain) statement()             {}
-func (*Grant) statement()               {}
-func (*Insert) statement()              {}
-func (*ParenSelect) statement()         {}
-func (*RenameColumn) statement()        {}
-func (*RenameDatabase) statement()      {}
-func (*RenameIndex) statement()         {}
-func (*RenameTable) statement()         {}
-func (*Revoke) statement()              {}
-func (*RollbackTransaction) statement() {}
-func (*Select) statement()              {}
-func (*Set) statement()                 {}
-func (*SetTransaction) statement()      {}
-func (*Show) statement()                {}
-func (*ShowColumns) statement()         {}
-func (*ShowDatabases) statement()       {}
-func (*ShowGrants) statement()          {}
-func (*ShowIndex) statement()           {}
-func (*ShowTables) statement()          {}
-func (*Truncate) statement()            {}
-func (*Union) statement()               {}
-func (*Update) statement()              {}
-func (Values) statement()               {}
+// StatementType implements the Statement interface.
+func (*AlterTable) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*BeginTransaction) StatementType() StatementType { return Ack }
+
+// StatementType implements the Statement interface.
+func (*CommitTransaction) StatementType() StatementType { return Ack }
+
+// StatementType implements the Statement interface.
+func (*CreateDatabase) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*CreateIndex) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*CreateTable) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*Delete) StatementType() StatementType { return RowsAffected }
+
+// StatementType implements the Statement interface.
+func (*DropDatabase) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*DropIndex) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*DropTable) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*Explain) StatementType() StatementType { return Rows }
+
+// StatementType implements the Statement interface.
+func (*Grant) StatementType() StatementType { return DDL }
+
+// StatementType implements the Statement interface.
+func (*Insert) StatementType() StatementType { return RowsAffected }
```

Comment:
How is this going to interact if we implement `INSERT...RETURNING`? With the RETURNING clause the statement show have a Rows-style result. Without the RETURNING clause it should have a RowsAffected-style result. We can certainly punt this concern to the future. Just curious if you had thoughts about it.